### ISSUE TITLE
Result changes if we don't pass attension mask in TFDistilbert model on SQUADv1 dataset

### ISSUE THREAD

**Issue by bhadreshpsavani** [huggingface/transformers#8680]
## Environment info
- `transformers` version: latest
- Platform: Colab
- Python version: 
- Tensorflow version (GPU?): 2.3.0
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help
I used the below code for getting Model
```
from transformers import  AutoTokenizer, TFAutoModelForQuestionAnswering
tokenizer = AutoTokenizer.from_pretrained('distilbert-base-uncased-distilled-squad')
model =  TFAutoModelForQuestionAnswering.from_pretrained('distilbert-base-uncased-distilled-squad', return_dict=True)
```
 tokenizers: @mfuntowicz
 examples/seq2seq: @patil-suraj
 tensorflow: @jplu 

## Information

The model I am using **TFDistilbert** pretrained.

The problem arises when using:
* my own modified scripts:
This is the Notebook
[Colab](https://github.com/bhadreshpsavani/UnderstandingNLP/blob/master/TFLiteExperimentsQALatest.ipynb)

The tasks I am working on is:
*  an official SQUaD task

## To reproduce

Steps to reproduce the behavior: [Colab](https://github.com/bhadreshpsavani/UnderstandingNLP/blob/master/TFLiteExperimentsQALatest.ipynb)

## Expected behavior

The performance should be the same because the Attention mask is the optional argument, if we don't pass it will create it internally. 

With Attention Mask:
```
OrderedDict([('exact', 77.71050141912077),
           ('f1', 85.5370981182013),
           ('total', 10570)])
```
Without Attention Mask:
```
OrderedDict([('exact', 72.82876064333927),
             ('f1', 80.71521545953475),
             ('total', 10570)])
```

**Comment by patrickvonplaten** [comment]
`attention_mask` is an optional argument, but that doesn't mean that it should not be passed to the function. If `attention_mask` is `None` then it is initialized to attend all tokens, which is incorrect if the input is a batch that include padding tokens. => It's better to simply pass the `attention_mask` to the forward function.
